users/newgrp.rs

```diff
@@ -0,0 +1,849 @@
+//
+// Copyright (c) 2024 Hemi Labs, Inc.
+//
+// This file is part of the posixutils-rs project covered under
+// the MIT License.  For the full license text, please see the LICENSE
+// file in the root directory of this project.
+// SPDX-License-Identifier: MIT
+//
+
+use clap::{error::ErrorKind, Parser};
+use gettextrs::{bind_textdomain_codeset, setlocale, textdomain, LocaleCategory};
+use libc::{
+    getgid, getgrnam, getgroups, getlogin, getpwnam, getpwuid, getuid, gid_t, passwd, setegid,
+    setgid, setgroups, setuid, uid_t,
+};
+
+#[cfg(target_os = "linux")]
+use libc::{ECHO, ECHONL, TCSANOW};
+#[cfg(target_os = "linux")]
+use libcrypt_rs::Crypt;
+use plib::{group::Group, PROJECT_NAME};
+
+use std::{
+    env,
+    ffi::{CStr, CString},
+    io,
+    process::{self, Command},
+};
+
+#[cfg(target_os = "linux")]
+use std::{
+    fs::File,
+    io::{BufRead, BufReader},
+    os::unix::io::AsRawFd,
+};
+
+#[cfg(target_os = "linux")]
+const GROUPSHADOW_PATH: &str = "/etc/gshadow";
+
+#[cfg(target_os = "linux")]
+const MAX_GROUPS: usize = libc::KERN_NGROUPS_MAX as usize;
+
+#[cfg(target_os = "macos")]
+const MAX_GROUPS: usize = 16;
+
+/// newgrp — change to a new group
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about)]
+struct Args {
+    /// Change the environment to what would be expected if the user actually logged in again (letter `l`).
+    #[arg(short = 'l')]
+    login: bool,
+
+    /// Specifies the group ID or group name. This is a positional argument that must be provided.
+    #[arg(value_name = "GROUP", required = true)]
+    group: String,
+}
```

Comment:
See datetime/cal.rs for an example of the new style of `struct Args` with translation support added.

Follow cal.rs style of struct Args.